src/components/TodoApp.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useState } from 'react';
+import { TodoList } from './TodoList';
+import { useTodoContext } from '../context/TodosContext';
+import { Filter } from './Filter';
+
+export const TodoApp: React.FC = () => {
+  const [query, setQuery] = useState<string>('');
+  const [activeCheckbox, secActiveCheckbox] = useState(false);
+
+  const { orderItems, setOrderItems } = useTodoContext();
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+  };
+
+  const handleQuerySubmit = useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+
+      if (query.trim().length <= 0) {
+        return;
+      }
+
+      setOrderItems(prevItems => {
+        const newItem = {
+          id: +new Date(),
+          title: query,
+          completed: false,
+          editing: false,
+        };
+
+        return [...prevItems, newItem];
+      });
+
+      setQuery('');
+    },
+    [query, setOrderItems, setQuery],
+  );
+
+  const handleActiveAllCheckbox = () => {
+    secActiveCheckbox(!activeCheckbox);
```

Comment:
- The same
- And actually you don't need to store this in state as it's calculated value